components/gitpod-db/src/typeorm/migration/1652365883273-CloneUrlIndexed.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists, indexExists } from "./helper/helper";
+
+export class CloneUrlIndexed1652365883273 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const TABLE_NAME = "d_b_workspace";
+        const COLUMN_NAME = "cloneURL";
+        const TYPE_INDEX_NAME = "d_b_workspace_cloneURL_idx";
+
+        if (!(await columnExists(queryRunner, TABLE_NAME, COLUMN_NAME))) {
+            await queryRunner.query(
+                `ALTER TABLE
+                    ${TABLE_NAME}
+                ADD COLUMN
+                    ${COLUMN_NAME} VARCHAR(256)
+                GENERATED ALWAYS AS (
```

Comment:
Nice! 😎 

Just read on it in https://dev.mysql.com/blog-archive/indexing-json-documents-via-virtual-columns/ and was wondering if `Stored Generated Columns – 3)` may apply here as well.

> [...] and you anticipate large amounts of scanning

That definitely applies here.